### PR TITLE
[Serializer] Rework `XmlEncoderTest`

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | 
| License       | MIT

I moved to use a data provider instead of extra test methods.

Would you accept this PR?
If yes, I will finish it

### CODE REVIEW

**Comment by alexandre-daubois** [comment]
That's nice! If this get accepted, don't forget to declare your new data providers static, a few ones are missing in the current codebase and will be fixed here https://github.com/symfony/symfony/pull/54824 🙂

File: src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php [hunk 1]
```diff
@@ -979,7 +922,7 @@ public function testEncodeWithoutPi()
 
         $expected = '<response/>';
 
-        $this->assertEquals($expected, $encoder->encode([], 'xml'));
+        static::assertEquals($expected, $encoder->encode([], 'xml'));
```
Comment (fabpot) [inline src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php]: As we're using `$this->` everywhere, this change should be part of PHP-CS-Fixer so that it can be enforced. Until then, I propose to revert such changes.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php [hunk 2]
```diff
@@ -110,83 +105,65 @@ public function testAttributes()
...
-        $this->assertSame($expected, $this->encoder->encode($array, 'xml', $context));
-    }
-
-    public function testEncodeNotRemovingEmptyTags()
-    {
-        $array = ['person' => ['firstname' => 'Peter', 'lastname' => null]];
-
-        $expected = '<?xml version="1.0"?>'."\n".
-            '<response><person><firstname>Peter</firstname><lastname/></person></response>'."\n";
+        yield 'encode remvoing empty tags' => [
```
Comment (mtarld) [inline src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php:152]:
```suggestion
        yield 'encode removing empty tags' => [
```

**Review by mtarld — APPROVED** [review]

**Comment by OskarStark** [comment]
Ready to merge from my side

File: src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php [hunk 3]
```diff
@@ -39,46 +39,39 @@ protected function setUp(): void
...
+    public function testEncode(string $expected, mixed $data, array $context = [])
     {
-        $obj = new ScalarDummy();
-        $obj->xmlFoo = 'foo';
-
-        $expected = '<?xml version="1.0"?>'."\n".
-            '<response>foo</response>'."\n";
-
-        $this->assertEquals($expected, $this->encoder->encode($obj, 'xml'));
+        static::assertSame($expected, $this->encoder->encode($data, 'xml', $context));
```
Comment (xabbuh) [inline src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php]: Why this change? We don't call the assertion methods in a static manner anywhere else in the tests, do we?

Comment (OskarStark) [reply src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php]: Damn, I replaced that already, as Fabien also added that feedback, not sure why its back again

Comment (OskarStark) [reply src/Symfony/Component/Serializer/Tests/Encoder/XmlEncoderTest.php]: Fixed

**Comment by OskarStark** [comment]
Rebased

**Comment by OskarStark** [comment]
Rebased

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @OskarStark.
